cobra/medium/boundary_types.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+
+"""Contains function to identify the type of boundary reactions.
+
+This module uses various heuristics to decide whether a boundary reaction
+is an exchange, demand or sink reaction. It mostly orientates on the following
+paper:
+
+Thiele, I., & Palsson, B. Ø. (2010, January). A protocol for
+generating a high-quality genome-scale metabolic reconstruction.
+Nature protocols. Nature Publishing Group.
+http://doi.org/10.1038/nprot.2009.203
+"""
+
+from collections import Counter
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+excludes = {"demand": ["SN_", "SK_", "sink", "EX_", "exchange"],
+            "exchange": ["demand", "DM_", "biosynthesis", "transcription",
+                         "replication", "SN_", "SK_", "sink"],
+            "sink": ["demand", "DM_", "biosynthesis", "transcription",
+                     "replication", "EX_", "exchange"]}
+"""A list of sub-strings in reaction IDs that usually indicate
+that the reaction is *not* a reaction of the specified type."""
+
+sbo_terms = {"demand": "SBO:0000628",
+             "exchange": "SBO:0000627",
+             "sink": "SBO:0000632",
+             "biomass": "SBO:0000629",
+             "pseudoreaction": "SBO:0000631"}
+"""SBO term identifiers for various boundary types."""
+
+
+def external_compartment(model):
+    """Find the external compartment in the model.
+
+    Uses a simple heuristic where the external compartment should be the one
+    with the most exchange reactions.
+
+    Arguments
+    ---------
+    model : cobra.Model
+        A cobra model.
+
+    Returns
+    -------
+    str
+        The putative external compartment.
+    """
+    counts = Counter(tuple(r.compartments)[0] for r in model.boundary)
+    most = counts.most_common(1)[0][0]
+    if "e" in model.compartments:
+        if most == "e":
+            return "e"
+        else:
+            LOGGER.warn("There is an `e` compartment but it does not look "
+                        "like it is the actual external compartment.")
+        return most
+    return most
+
+
+def is_type(reaction, btype, ext_compartment):
```

Comment:
I would like the publicly visible names to be a little more verbose. In this case, for example, I think `is_boundary_type` is more descriptive. Similarly the arguments could be `boundary_type` and `external_compartment`.